salt/cloud/__init__.py

```diff
@@ -1494,7 +1494,7 @@ def do_action(self, names, kwargs):
                         else:
                             log.debug(
                                 'vm:{0} in provider:{1} is not in name '
-                                'list:\'{2}\''.format(vm_name, driver, names)
+                                'list:\'{2}\''.format(vm_name.encode('utf-8'), driver, names)
```

Comment:
This should likely be `__salt_system_encoding__` instead of `utf-8`. That will select the correct encoding type.

Reply:
This is the wrong fix.
from __future__ import unicode_literals
   OR
prefix all strings with u'string'

The reason this is broken is discussed in this large issue https://github.com/saltstack/salt/issues/42278#issuecomment-318913952

The long term fix is to ensure everything is in Unicode Format i.e. salt.cloud.CloudClient always returns Unicode no matter what the driver is and takes Unicode and the drive converts down from Unicode if it needs to.

The fix I suggest allows logging to work with mixed PY2 and PY2 Unicode.

Reply:
@konstest Can you change this to `__salt_system_encoding__` as @cachedout suggested above? The other comments from @damon-atkins are being handled in a different way right now.

Once that is updated, we can get this fix in.